src/components/NetworkOverlay/index.tsx

```diff
@@ -103,7 +109,28 @@ export default function NetworkOverlay() {
 
   return (
     <>
-      {isConnected && chainId?.toString() !== '100' && (
+      {loading && (
```

Comment:
Like 
```suggestion
return (
    <>
      {
        ()=>{if(loading) {
          return(
            <Overlay
              className={'OverlayWrongNetwork'}
            >
              <style>{css}</style>
              <CircularProgress/>
            </Overlay>
          )
        }}
      }
    </>
);
```

instead of 

```
return (
    <>
      {loading && (
        <Overlay
          className={'OverlayWrongNetwork'}
        >
          <style>{css}</style>
          <CircularProgress/>
        </Overlay>
      )}
    </>
  );
  ```
  ?

Reply:
changed